test/c/CMakeLists.txt

```
@@ -19,12 +19,13 @@ file (
 # test file.
 create_test_sourcelist (source_files test_driver.c ${discovered_tests})
 
+add_library(common common.c)
```

So I'm not sure if this fixes it on windows (locally it still works on my mac) but it seems like windows is not finding the library (again), maybe this fixes it?
```suggestion
add_library(common common.c)
install(TARGETS common RUNTIME DESTINATION ${CMAKE_BINARY_DIR})
```
which should install the dll into the path of the executable. We could also consider putting this inside a `if (MSVC)` block, though it didn't seem to affect running tests for me.